### PR TITLE
Add Intl.getCanonicalLocales

### PR DESCRIPTION
**Pull request by pmarkee**
Signed-off-by: Peter Marki marpeter@inf.u-szeged.hu

### CODE REVIEW

**Comment by pmarkee** [comment]
Related issue: #251

**Comment by pmarkee** [comment]
Depends on #260 for testing.

File: src/runtime/GlobalObjectBuiltinIntl.cpp [hunk 1]
```diff
@@ -2539,6 +2539,18 @@ static Value builtinIntlNumberFormatSupportedLocalesOf(ExecutionState& state, Va
...
 }
 
+static Value builtinIntlGetCanonicalLocales(ExecutionState& state, Value thisValue, size_t argc, Value* argv, bool isNewExpression)
+{
+    // Let ll be ? CanonicalizeLocaleList(locales).
+    ValueVector ll = canonicalizeLocaleList(state, argv[0]);
+    // Return CreateArrayFromList(ll);
+    ArrayObject* ret = new ArrayObject(state, 0.0);
+    for (size_t i = 0; i < ll.size(); i++) {
+        ret->defineOwnProperty(state, ObjectPropertyName(state, Value(i)), ObjectPropertyDescriptor(Value(ll[i]), ObjectPropertyDescriptor::AllPresent));
```
Comment (clover2123) [inline src/runtime/GlobalObjectBuiltinIntl.cpp]: For `Value(ll[i])`, do we have to generate a temporal Value? How about directly passing the element like `ObjectPropertyDescriptor(ll[i], ...)`?

Comment (pmarkee) [reply src/runtime/GlobalObjectBuiltinIntl.cpp]: Indeed we can pass it like `ObjectPropertyDescriptor(ll[i], ...)`, thanks.

File: src/runtime/GlobalObjectBuiltinIntl.cpp [hunk 2]
```diff
@@ -2539,6 +2539,18 @@ static Value builtinIntlNumberFormatSupportedLocalesOf(ExecutionState& state, Va
     return supportedLocales(state, availableLocales, requestedLocales, options);
 }
 
+static Value builtinIntlGetCanonicalLocales(ExecutionState& state, Value thisValue, size_t argc, Value* argv, bool isNewExpression)
+{
```
Comment (yichoi) [inline src/runtime/GlobalObjectBuiltinIntl.cpp]: Check pointer-type parameter / check before deferencing
ASSERT(argv != nullptr);

**Comment by pmarkee** [comment]
@yichoi can you please check this PR? Thanks.

**Pull request merged**